nexus/src/app/rack.rs

```diff
@@ -304,139 +374,140 @@ impl super::Nexus {
+                };
+
+                let mut port_settings_params = SwitchPortSettingsCreate {
+                    identity,
+                    port_config,
+                    groups: vec![],
+                    links: HashMap::new(),
+                    interfaces: HashMap::new(),
+                    routes: HashMap::new(),
+                    bgp_peers: HashMap::new(),
+                    addresses: HashMap::new(),
+                };
+
+                let uplink_address = IpNet::from_str(&format!("{}/32", uplink_config.uplink_ip))
+                    .map_err(|e| Error::internal_error(&format!(
                     "failed to parse value provided for `rack_network_config.uplink_ip` as `IpNet`: {e}"
                 )))?;
 
-            let address = Address {
-                address_lot: NameOrId::Name(address_lot_name),
-                address: uplink_address,
-            };
-            port_settings_params.addresses.insert(
-                "phy0".to_string(),
-                AddressConfig { addresses: vec![address] },
-            );
-
-            let dst = IpNet::from_str("0.0.0.0/0").map_err(|e| {
-                Error::internal_error(&format!(
-                    "failed to parse provided default route CIDR: {e}"
-                ))
-            })?;
-
-            let gw = IpAddr::V4(rack_network_config.gateway_ip);
-            let vid = rack_network_config.uplink_vid;
-            let route = Route { dst, gw, vid };
+                let address = Address {
+                    address_lot: NameOrId::Name(address_lot_name.clone()),
+                    address: uplink_address,
+                };
+                port_settings_params.addresses.insert(
+                    "phy0".to_string(),
+                    AddressConfig { addresses: vec![address] },
+                );
 
-            port_settings_params.routes.insert(
-                "phy0".to_string(),
-                RouteConfig { routes: vec![route] },
-            );
-
-            if self
-                .db_datastore
-                .switch_port_settings_get(opctx, &name.into())
-                .await
-                .is_err()
-            {
-                self.db_datastore
-                    .switch_port_settings_create(opctx, &port_settings_params)
-                    .await?;
-            }
+                let dst = IpNet::from_str("0.0.0.0/0").map_err(|e| {
+                    Error::internal_error(&format!(
+                        "failed to parse provided default route CIDR: {e}"
+                    ))
+                })?;
 
-            // TODO - https://github.com/oxidecomputer/omicron/issues/3277
-            // record port speed
+                let gw = IpAddr::V4(uplink_config.gateway_ip);
+                let vid = uplink_config.uplink_vid;
+                let route = Route { dst, gw, vid };
+
+                port_settings_params.routes.insert(
+                    "phy0".to_string(),
```

Comment:
Should the name of this correspond to `uplink_port`? e.g. if they choose `qsfp12` instead of `qsfp0`

Reply:
phy0 here refers to the first physical lane of the port. When we gain support for breakout cables this will no longer be hard coded.